app/main.py

```diff
@@ -171,9 +171,15 @@ def _create_kubernetes_secret_objects(self, key, secret_value, secret_type):
         if secret.type == 'kubernetes.io/tls':
             _logger.info('Extracting private key and certificate.')
             p12 = crypto.load_pkcs12(base64.decodestring(secret_value))
+            if os.getenv('DOWNLOAD_CA_CERTIFICATES','true').lower() == "true":
```

Comment:
I am not sure you need this check. when would you ever want to download the cert without the root CA cert?

Reply:
It is an existing feature, so I kept it as is.

Reply:
We came across at least one client that failed to work when the intermediate certificates were included. That's why I added this setting (which is off by default)